include/xtypes/DynamicDataImpl.hpp

```diff
@@ -274,318 +274,424 @@ inline DynamicData DynamicData::operator - () const
     return *this;
 }
 
-#define DYNAMIC_DATA_OPERATOR_RESULT(TYPE, OPERATOR) \
-{\
-    TYPE lho = *this;\
-    TYPE rho = other;\
-    DynamicData result(primitive_type<TYPE>());\
-    result = lho OPERATOR rho;\
-    return result;\
-}
-
-inline DynamicData DynamicData::operator * (const ReadableDynamicDataRef& other) const
+inline DynamicData DynamicData::operator ~ () const
 {
     switch(type_.kind())
     {
         case TypeKind::INT_8_TYPE:
-            DYNAMIC_DATA_OPERATOR_RESULT(int8_t, *);
+            DYNAMIC_DATA_UNARY_OPERATOR_RESULT(int8_t, ~);
         case TypeKind::UINT_8_TYPE:
-            DYNAMIC_DATA_OPERATOR_RESULT(uint8_t, *);
+            DYNAMIC_DATA_UNARY_OPERATOR_RESULT(uint8_t, ~);
         case TypeKind::INT_16_TYPE:
-            DYNAMIC_DATA_OPERATOR_RESULT(int16_t, *);
+            DYNAMIC_DATA_UNARY_OPERATOR_RESULT(int16_t, ~);
         case TypeKind::UINT_16_TYPE:
-            DYNAMIC_DATA_OPERATOR_RESULT(uint16_t, *);
+            DYNAMIC_DATA_UNARY_OPERATOR_RESULT(uint16_t, ~);
         case TypeKind::INT_32_TYPE:
-            DYNAMIC_DATA_OPERATOR_RESULT(int32_t, *);
+            DYNAMIC_DATA_UNARY_OPERATOR_RESULT(int32_t, ~);
         case TypeKind::UINT_32_TYPE:
-            DYNAMIC_DATA_OPERATOR_RESULT(uint32_t, *);
+            DYNAMIC_DATA_UNARY_OPERATOR_RESULT(uint32_t, ~);
         case TypeKind::INT_64_TYPE:
-            DYNAMIC_DATA_OPERATOR_RESULT(int64_t, *);
+            DYNAMIC_DATA_UNARY_OPERATOR_RESULT(int64_t, ~);
         case TypeKind::UINT_64_TYPE:
-            DYNAMIC_DATA_OPERATOR_RESULT(uint64_t, *);
-        case TypeKind::FLOAT_32_TYPE:
-            DYNAMIC_DATA_OPERATOR_RESULT(float, *);
-        case TypeKind::FLOAT_64_TYPE:
-            DYNAMIC_DATA_OPERATOR_RESULT(double, *);
-        case TypeKind::FLOAT_128_TYPE:
-            DYNAMIC_DATA_OPERATOR_RESULT(long double, *);
+            DYNAMIC_DATA_UNARY_OPERATOR_RESULT(uint64_t, ~);
         default:
-            xtypes_assert(false,
-                "operator*() isn't available for types '" << type_.name() << "' and '" << other.type().name() << "'.");
-            return *this;
+            xtypes_assert(false, "operator~() isn't available for type '" << type_.name() << "'.");
+            break;
     }
+
+    return *this;
 }
 
-inline DynamicData DynamicData::operator / (const ReadableDynamicDataRef& other) const
+#define DYNAMIC_DATA_NOT_OPERATOR_RESULT(TYPE) \
+{\
+    return !bool(this->ReadableDynamicDataRef::value<TYPE>());\
+}
+
+#define DYNAMIC_DATA_SELF_DE_INCREMENT(TYPE, OPERATOR) \
+{\
+    TYPE pre = *this; \
+    this->WritableDynamicDataRef::value<TYPE>(OPERATOR(pre)); \
```

Comment:
This should be equivalent, isn't?
```suggestion
    this->value<TYPE>(OPERATOR(pre)); \
```